src/cascadia/TerminalControl/ThrottledFunc.h

```diff
@@ -1,184 +1,76 @@
-/*++
-Copyright (c) Microsoft Corporation
-Licensed under the MIT license.
-
-Module Name:
-- ThrottledFunc.h
---*/
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
 
 #pragma once
-#include "pch.h"
 
-template<typename... Args>
-class ThrottledFuncStorage
+template<bool leading, typename... Args>
+class ThrottledFunc
 {
-public:
-    template<typename... MakeArgs>
-    bool Emplace(MakeArgs&&... args)
-    {
-        std::scoped_lock guard{ _lock };
-
-        const bool hadValue = _pendingRunArgs.has_value();
-        _pendingRunArgs.emplace(std::forward<MakeArgs>(args)...);
-        return hadValue;
-    }
-
-    template<typename F>
-    void ModifyPending(F f)
-    {
-        std::scoped_lock guard{ _lock };
-
-        if (_pendingRunArgs.has_value())
-        {
-            std::apply(f, _pendingRunArgs.value());
-        }
-    }
+    using impl = til::throttled_func<leading, Args...>;
 
-    std::tuple<Args...> Extract()
-    {
-        decltype(_pendingRunArgs) args;
-        std::scoped_lock guard{ _lock };
-        _pendingRunArgs.swap(args);
-        return args.value();
-    }
-
-private:
-    std::mutex _lock;
-    std::optional<std::tuple<Args...>> _pendingRunArgs;
-};
-
-template<>
-class ThrottledFuncStorage<>
-{
 public:
-    bool Emplace()
-    {
-        return _isRunPending.test_and_set(std::memory_order_relaxed);
-    }
-
-    std::tuple<> Extract()
+    ThrottledFunc(
+        winrt::Windows::UI::Core::CoreDispatcher dispatcher,
+        typename impl::filetime_duration delay,
+        typename impl::function func) :
+        _impl{
+            delay,
+            [this, dispatcher = std::move(dispatcher), func = std::move(func)](auto&&... args) {
+                _dispatching.store(true, std::memory_order_relaxed);
```

Comment:
I just realized that this can lead to a deadlock under certain conditions.
I'll fix the code very soon. The 3 new files in "til" should be fine though.